xonsh/environ.py

```diff
@@ -454,8 +457,9 @@ def multiline_prompt():
     'TITLE': DEFAULT_TITLE,
     'MULTILINE_PROMPT': '.',
     'XONSHRC': os.path.expanduser('~/.xonshrc'),
-    'XONSH_HISTORY_SIZE': 8128,
-    'XONSH_HISTORY_FILE': os.path.expanduser('~/.xonsh_history'),
+    'XONSH_HISTORY_SIZE': (8128, 'files'),
+    'XONSH_HISTORY_FILE': os.path.expanduser('~/.xonsh_history.json'),
+    'XONSH_STORE_STDOUT': True,
```

Comment:
I might argue that this should default to `False`, since I'm not sure it will be used terribly often and it has the potential to take up a lot of disk space.  What do you think, @scopatz?